src/libs/IrcClient.js

```diff
@@ -413,12 +413,19 @@ function clientMiddleware(state, network) {
             }
 
             let buffer = state.getOrAddBufferByName(networkid, event.channel);
-            state.addUserToBuffer(buffer, {
+            let user = {
                 nick: event.nick,
                 username: event.ident,
                 host: event.hostname,
                 realname: event.gecos,
-            });
+            };
+
+            // Add account to the user object if extended-join is enabled
+            if (client.network.cap.isEnabled('extended-join')) {
```

Comment:
This could simply be `if (user.account) {` so that it always adds the account if it's available